python/marvin/tools/rss.py

```diff
@@ -1,102 +1,116 @@
+    """A class to interface with a MaNGA DRP row-stacked spectra file.
 
+    This class represents a fully reduced DRP row-stacked spectra object,
+    initialised either from a file, a database, or remotely via the Marvin API.
+    Instances of `.RSS` are a list of `.RSSFiber` objects, one for each fibre
+    and exposure. `.RSSFiber` are initialised lazily, containing only basic
+    information. They need to be initialised by calling `.RSSFiber.load`
+    (unless `.RSS.autoload` is ``True``, in which case the instance is loaded
+    when first accessed).
 
-class RSS(MarvinToolsClass, list):
-    """A class to interface with MaNGA RSS data.
-
-    This class represents a fully reduced RSS file, initialised either
-    from a file, a database, or remotely via the Marvin API. The class
-    inherits from Python's list class, and is defined as a list of
-    RSSFiber objects.
-
-    Parameters:
-        filename (str):
-            The path of the file containing the RSS to load.
-        mangaid (str):
-            The mangaid of the RSS to load.
-        plateifu (str):
-            The plate-ifu of the RSS to load (either ``mangaid`` or
-            ``plateifu`` can be used, but not both).
-        mode ({'local', 'remote', 'auto'}):
-            The load mode to use. See
-            :doc:`Mode secision tree</mode_decision>`.
-        nsa_source ({'auto', 'drpall', 'nsa'}):
-            Defines how the NSA data for this object should loaded when
-            ``RSS.nsa`` is first called. If ``drpall``, the drpall file will
-            be used (note that this will only contain a subset of all the NSA
-            information); if ``nsa``, the full set of data from the DB will be
-            retrieved. If the drpall file or a database are not available, a
-            remote API call will be attempted. If ``nsa_source='auto'``, the
-            source will depend on how the ``RSS`` object has been
-            instantiated. If the cube has ``RSS.data_origin='file'``,
-            the drpall file will be used (as it is more likely that the user
-            has that file in their system). Otherwise, ``nsa_source='nsa'``
-            will be assumed. This behaviour can be modified during runtime by
-            modifying the ``RSS.nsa_mode`` with one of the valid values.
-        release (str):
-            The MPL/DR version of the data to use.
-
-    Return:
-        rss:
-            An object representing the RSS entity. The object is a list of
-            RSSFiber objects, one for each fibre in the RSS entity.
+    In addition to the input arguments supported by `~.MarvinToolsClass` and
+    `~.NSAMixIn`, this class accepts an ``autoload`` keyword argument that
+    defines whether `.RSSFiber` objects should be automatically loaded when
+    they are accessed.
 
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, input=None, filename=None, mangaid=None, plateifu=None,
```

Comment:
I think we want to add connections between `Cube` and `RSS` or maybe `Spaxel` and `RSS` a la the `getXXX` methods.  

Also since you're opening RSS only via files, do we want to include more information from the `OBSINFO` extension somewhere within the object?  In general if we include this in our Cubes or RSS we can display/provide information about the observing conditions somewhere.

Reply:
Something I forgot is to add the relevant lines from the `OBSINFO` to each `RSSFiber`. I'll add a `getCube` link. I feel that's enough and we can go from there to the Maps/ModelCubes. I don't feel a need to overcrowd the namespace with links in all directions, but I can be convienced otherwise.

I don't think we want to include `OBSINFO` information in the `Cube`, that seems a bit too much cross talk. If we ever implement the contributors to an spaxel thing that may be a way to connect the spaxels to the obsinfo in their corresponding fibres. 

There may be some columns in that table that are important enough that we want to expose the in some way, but I'd leave that for future enhancements.